src/data/nat/totient.lean

```diff
@@ -335,4 +335,33 @@ begin
   rw [sub_mul, one_mul, mul_comm, mul_inv_cancel hp', cast_pred hp],
 end
 
+lemma gcd_prod_double_counting {β : Type*} [comm_monoid β] (m n : ℕ) (f : ℕ → β) :
```

Comment:
`prod_factors_gcd_mul_prod_factors_mul` (+additive version)?

Reply:
Hmm, `data/nat/prime` doesn't know about `prod_union_inter`, which is defined in `algebra/big_operators/basic`, while `data/nat/gcd` doesn't know about `factors`.